base/sparse/sparsevector.jl

```diff
@@ -1616,3 +1617,15 @@ function _densifystarttolastnz!(x::SparseVector)
     x.nzind[1:newnnz] = 1:newnnz
     x
 end
+
+#sorting
+function sort{Tv,Ti}(x::SparseVector{Tv,Ti}; kws...)
+    allvals = push!(copy(nonzeros(x)),zero(Tv))
+    sinds = sortperm(allvals;kws...)
+    n,k = length(x),length(allvals)
+    z = findfirst(sinds,k)
+    newnzind = collect(Ti,1:k-1)
+    newnzind[z:end]+= n-k+1
+    newnzvals = allvals[sinds[deleteat!(collect(1:k),z)]]
```

Comment:
Probably won't cause in huge speedups, but `sinds[deleteat!(collect(1:k),z)]` can be changed to `deleteat!(sinds[1:k], z)` to avoid a temporary array allocation here.